pkg/channel/event_dispatcher.go

```diff
@@ -80,6 +98,19 @@ func (d *EventDispatcher) DispatchEvent(ctx context.Context, event cloudevents.E
 		destinationURL := d.resolveURL(destination)
 		ctx, response, err = d.executeRequest(ctx, destinationURL, event)
 		if err != nil {
+
+			if delivery != nil && delivery.DeadLetterSink != "" {
+				deadLetterURL := d.resolveURL(delivery.DeadLetterSink)
```

Comment:
Do we want to resolve this each time there's a failure?

Reply:
Probably not but notice we also resolve the destination and the reply for each request. Let's fix it later on?